### PR TITLE
Enhancement: added cancel sync button in sync progress dialog

### LINKED ISSUE #12613: [Bug] Clicking outside of sync window cancels sync without cleanly disconnecting from sync service
###### Reproduction Steps

1. Using beta86, review some cards and then click the sync button 
2. Wait for syncing dialog to appear
3. Click anywhere outside of syncing progress window

###### Expected Result

I wouldn't expect clicking on the screen to have any immediate action on the syncing. Maybe it could ask me if I wanted to cancel syncing, or instead of showing the detailed syncing progress by default it could animate the sync icon and create a notification instead of a window and I could interact with the notification to review/cancel progress of the sync.

Also, when double checking this before submitting the bug report, I got a crash after doing the above steps (not reproducible)

###### Actual Result

A snackbar immediately pops up saying "Interrupted" and further attempts to sync give an error saying "only one copy of anki can sync at a time, try again later"

###### Debug info

```
AnkiDroid Version = 2.16alpha86

Android Version = 12

Manufacturer = samsung

Model = SM-G970U

Hardware = qcom

Webview User Agent = Mozilla/5.0 (Linux; Android 12; SM-G970U Build/SP1A.210812.016; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/105.0.5195.136 Mobile Safari/537.36

ACRA UUID = f9d003f3-ae34-4529-8cd5-ed8aff42fa30

New schema = true

Scheduler = std3

Crash Reports Enabled = true

DatabaseV2 Enabled = true
```

###### Research
*Enter an [x] character to confirm the points below:*

- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
- [x] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### PR DESCRIPTION
**Pull request by criticalAY**
## Purpose / Description
Add cancel button in sync progress dialog 

## Fixes
* Fixes #12613

## How Has This Been Tested?
Oneplus nord ce

https://github.com/ankidroid/Anki-Android/assets/48384865/aaebe4ab-71f3-4799-89fc-ce4fa385c98a

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/CoroutineHelpers.kt [hunk 1]
```diff
@@ -319,12 +322,20 @@ suspend fun <T> withProgressDialog(
...
+    showCancellable: Boolean = false,
     op: suspend (android.app.ProgressDialog) -> T
 ): T = coroutineScope {
     val dialog = android.app.ProgressDialog(context, R.style.AppCompatProgressDialogStyle).apply {
         setCancelable(onCancel != null)
-        onCancel?.let {
-            setOnCancelListener { it() }
+        if (showCancellable) {
+            setCancelable(false)
+            setButton(DialogInterface.BUTTON_NEGATIVE, context.getString(R.string.cancel_sync)) { _, _ ->
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/CoroutineHelpers.kt:332]: This isn't generic enough

`@StringRes manualCancelButton: Int = null` would probably be what we're looking for

The act of passing in the resource as non-null would act as `showCancellable` here

**Review by david-allison — CHANGES_REQUESTED** [review]
One issue with the  field not being generic enough

I would love for the back button to be handled, but we could leave this as a TODO

File: AnkiDroid/src/main/java/com/ichi2/anki/CoroutineHelpers.kt [hunk 2]
```diff
@@ -264,7 +264,7 @@ suspend fun <T> Backend.withProgress(
 suspend fun <T> FragmentActivity.withProgress(
     extractProgress: ProgressContext.() -> Unit,
     onCancel: ((Backend) -> Unit)? = { it.setWantsAbort() },
-    showCancellable: Boolean = false,
+    manualCancelButton: Int? = null,
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/CoroutineHelpers.kt:267]: `@StringRes`?

**Review by david-allison — APPROVED** [review]
One nitpick, requested updated video

File: AnkiDroid/src/main/java/com/ichi2/anki/CoroutineHelpers.kt (hunk 2 above)
Comment (criticalAY) [reply AnkiDroid/src/main/java/com/ichi2/anki/CoroutineHelpers.kt:267]: oops sorry!

Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/CoroutineHelpers.kt:267]: Never a worry!

**Comment by criticalAY** [comment]
Updated test video :
https://github.com/ankidroid/Anki-Android/assets/48384865/896dcef3-78df-4040-967c-09392630a76f

**Review by david-allison — APPROVED** [review]
Only question is whether 'cancel' should be blue as that feels inconsistent

That's a nitpick of a nitpick, let's get this in, one step closer to 2.17!

**Comment by criticalAY** [comment]
> Only question is whether 'cancel' should be blue as that feels inconsistent

Can make it blue won't be an issue just need to change one word

**Comment by criticalAY** [comment]
Updated the color 
image

File: AnkiDroid/src/main/res/values/styles.xml [hunk 3]
```diff
@@ -193,6 +193,6 @@
     </style>
 
     <style name="CustomButtonBarButtonStyle" parent="Widget.AppCompat.Button.ButtonBar.AlertDialog">
-        <item name="android:textColor">?android:attr/textColor</item>
+        <item name="android:textColor">@color/material_light_blue_500</item>
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/res/values/styles.xml:196]: Plain theme doesn't use blue. Dialogs use `?attr/colorPrimary` IIRC

**Comment by mikehardy** [comment]
Hoping to merge this one today - two notes with regards to strings:
- strings are currently clear after merge of #15326 
- strings workflow just switched from node v18 to v20 in #15327 and this will be first run with that, so carefully watch the strings workflow

**Pull request merged**